indigo_api/models/timeline.py

```diff
@@ -0,0 +1,174 @@
+from django.utils.formats import date_format
+from django.utils.translation import gettext as _
+
+
+class TimelineEntry:
```

Comment:
Python`s dataclasses are really useful for something like this. It will handle defaults and setup a constructor that takes the various fields as parameters.

See https://docs.python.org/3/library/dataclasses.html and https://github.com/laws-africa/indigo-lawsafrica/blob/master/citator/extraction/patterns.py#L18